partner_lastname_uppercase/models/res_partner.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2023 Coop IT Easy SC
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.model
+    def uppercase_all_lastnames(self):
+        individuals = self.search(
+            [
+                ("is_company", "=", False),
+                ("lastname", "!=", False),
+            ]
+        )
+        for partner in individuals:
+            partner.lastname = partner.lastname.upper()
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            is_company = vals.get("is_company", False)
+            lastname = vals.get("lastname", False)
+            if lastname and not is_company:
+                vals["lastname"] = lastname.upper()
+        return super().create(vals_list)
+
+    def write(self, vals):
+        res = super().write(vals)
+        individuals = self.filtered(lambda rp: rp.lastname and not rp.is_company)
+        for partner in individuals:
+            super(ResPartner, partner).write({"lastname": partner.lastname.upper()})
+        return res
```

Comment:
this upper-cases the last name at each write of a partner instead of only when the last name is changed. there could be common code between `create()` and `write()` in handling `vals` (maybe not that feasible because of the `is_company` test).

Reply:
Indeed. `if "lastname" in vals` would be a good check to add.

I'm not certain it's valuable to spend time creating common code between `write` and `create`. The module's too small for there to be any benefits.

Reply:
I tend to agree with Carmen, I started doing common code which pretty much reduced to calling `upper` so I left it as is.